Prevent resetting LDFLAGS

This allows the plan9port build to inherit ldflags from the environment.

(This patch must be applied after plan9port-noexecstack.patch)
--- a/src/mkhdr
+++ b/src/mkhdr
@@ -12,7 +12,6 @@
 AS=9a
 AR=9ar
 CFLAGS=
-LDFLAGS=
 AFLAGS=--noexecstack
 CLEANFILES=
 NUKEFILES=
